src/bpftrace.cpp

```diff
@@ -1758,11 +1758,30 @@ std::string BPFtrace::get_stack(uint64_t stackidpid, bool ustack, StackType stac
                             stack_trace.data());
   if (err)
   {
-    // ignore EFAULT errors: eg, kstack used but no kernel stack
-    if (stackid != -EFAULT)
+    if (stackid == -EFAULT && stack_type.type == StackType::ukstack::kstack)
```

Comment:
Am I going crazy or should we be checking `err == -EFAULT` instead of `stackid`? It's been this way for years

Reply:
It certainly looks so, checking `stackid` doesn't make any sense.